Products/Zuul/routers/network.py

```diff
@@ -114,6 +117,46 @@ def deleteNode(self, uid):
         audit('UI.Network.DeleteSubnet', subnet=uid)
         return DirectResponse.succeed(tree=self.getTree())
 
+    def _marshal(self, netid=None, shift=0, limit=None):
+        root = self.catalog.search(IpNetwork, depth=0, paths=[netid]).results.next()
```

Comment:
If we know the id of the IpNetwork that is the root then a catalog search is unnecessary.